scripts/us_bls/jolts/BLSJolts.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+import urllib.request
+OUTPUT_MCF_TO_CONSOLE = True
+
+### Step 1) Data Download
+# Dataset being processed: https://download.bls.gov/pub/time.series/jt/
+# Data split across several white space separate files that may have incomplete rows
+def web_to_pandas(url, sep):
+  '''
+    Helper to load incomplete whitespace separated data file into pandas
+  '''
+  df = None
+  remote_web_page = urllib.request.urlopen(url)
+
+  for line in remote_web_page:
+    # Split out columns
+    line_text = line.decode("utf-8")
+    line_text = line_text.replace('\r', '')
+    line_text = line_text.replace('\n', '')
+    cols_in_line = [col.strip() for col in line_text.split(sep)]
+
+    # Initialize dataframe with first row
+    if df is None:
+      df = pd.DataFrame(columns = cols_in_line)
+
+    # Otherwise add a new row
+    else:
+      # If a row is missing a trailing column then pad it 
+      cols_in_line.extend((df.shape[1] - len(cols_in_line)) * [""])
+      df = df.append(pd.DataFrame([cols_in_line], columns=df.columns), ignore_index=True)
+
+  return df
+
+# Download series descriptions (used for seasonally adjusted status and industry code)
+series_desc = web_to_pandas("https://download.bls.gov/pub/time.series/jt/jt.series", sep="\t")
```

Comment:
Since the industry id is encoded in the series name,do you really need to use this file?  Are there any values in here that aren't in any of the specific data files. e.g if you ran unique on any of the specific series, would all the values be in it that are in jt.series?

Reply:
We pull some additional information from the series such as seasonality.